presto-main/src/main/java/com/facebook/presto/server/SimpleHttpResponseHandler.java

```diff
@@ -57,9 +58,14 @@ else if (response.getStatusCode() == HttpStatus.SERVICE_UNAVAILABLE.code()) {
                 callback.failed(new ServiceUnavailableException(uri));
             }
             else {
-                // Something is broken in the server or the client, so fail immediately (includes 500 errors)
                 Exception cause = response.getException();
-                if (cause == null) {
+                String responseErrorCode = response.getHeader("X-Presto-Error-Code");
```

Comment:
Wonder if this header should go into presto-client/src/main/java/com/facebook/presto/client/PrestoHeaders.java

Reply:
maybe we could have classes to capture internal headers. Usually we use headers only for our clients. But maybe from now on, we will see more use cases for Java/C++ communication.

Reply:
Sure, but we already use headers like PRESTO_MAX_WAIT in internal code, e.g. TaskResource.

Reply:
That is a good point. Maybe the same PrestoHeaders is good enough.

Reply:
Discussed offline with @tanjialiang, we will address this in another PR.